packages/dd-trace/src/config.js

```diff
@@ -701,6 +703,7 @@ class Config {
     this._envUnprocessed['appsec.wafTimeout'] = DD_APPSEC_WAF_TIMEOUT
     this._setBoolean(env, 'clientIpEnabled', DD_TRACE_CLIENT_IP_ENABLED)
     this._setString(env, 'clientIpHeader', DD_TRACE_CLIENT_IP_HEADER)
+    this._setBoolean(env, 'crashtracking.enabled', coalesce(DD_CRASHTRACKING_ENABLED, !!DD_INJECTION_ENABLED))
```

Comment:
I prefer having semantic dependencies between options be computed in `_applyCalculated()`. If implemented like that, here you'd only have:
```javascript
this._setBoolean(env, 'crashtracking.enabled', DD_CRASHTRACKING_ENABLED)
```
so `this._env.crashtracking.enabled` will unambiguously represent the value of that one environment variable. In `_applyCalculated()` you'd then have:
```javascript
if (this._env.injectionEnabled?.length > 0) {
  this._setBoolean(calc, 'crashtracking.enabled', true)
}
```
As a small sidenote I'm not using `this._env.crashtracking?.enabled || this._env.injectionEnabled?.length > 0` as the predicate above because calculated values are taken into consideration after environment variables in `_merge ()`, so if `this._env.crashtracking.enabled` is defined it'll be used first anyway.

Reply:
Good catch, and since this also affects telemetry I changed it 👍